src/python/dxpy/bindings/apollo/path_validator.py

```diff
@@ -57,12 +57,15 @@ def is_cohort_or_dataset(self):
         # since object is record:
         self.try_populate_record_http_request_info()
         if not (
-            ("Dataset" in self.record_http_request_info["recordTypes"])
-            or ("CohortBrowser" in self.record_http_request_info["recordTypes"])
+            ("Dataset" in self.record_http_request_info.get("recordTypes",[]))
+            or ("CohortBrowser" in self.record_http_request_info.get("recordTypes",[]))
+            or ("Dataset" in self.record_http_request_info.get("types",[]))
+            or ("CohortBrowser" in self.record_http_request_info.get("types",[]))
         ):
+            # Check recordTypes first, then types, and if neither have data, put TYPENOTFOUND in the type part of the string
             self.throw_error(
-                "Invalid path. The path must point to a record type of cohort or dataset and not a {} object."
-            ).format(self.record_http_request_info["recordTypes"])
+                "Invalid path. The path must point to a record type of cohort or dataset and not a {} object.".format(self.record_http_request_info.get("recordTypes",self.record_http_request_info.get("types","TYPENOTFOUND")))
```

Comment:
Very minor, but it could be also:

```suggestion
                "Invalid path. The path must point to a record type of cohort or dataset and not a {} object.".format(self.record_http_request_info.get("recordTypes") or self.record_http_request_info.get("types") or "TYPENOTFOUND")
```